controller/deliveryController.ts

```diff
@@ -3,7 +3,19 @@ import { deliveryEmployee } from "../model/deliveryEmployee"
 
 import {createDeliveryEmployee, getDeliveryEmployeeByID, getDeliveryEmployees} from "../service/deliveryService"
 
+const validateAccess = function (req: Request, res: Response, next: Function) {
+    if (req.session.current?.role === 'HR' || req.session.current?.role === 'SUPERUSER') {
+        next();
+    } else {
+        res.redirect('/forbidden');
+    }
+};
+
 export const deliveryController = (app:Application) =>{
+    app.use('/add-delivery-employee', validateAccess);
```

Comment:
You could do this on the get/post. For example:

` app.get('/add-delivery-employee', validateAccess, async (req:Request, res: Response) => {`

If you had the code in the function as a method, your routes start looking like 

` app.get('/add-delivery-employee', validateAccess, addDeliverEmployeeGet)`